consistent-sampling/src/main/java/io/opentelemetry/contrib/sampler/consistent2/ConsistentSampler.java

```diff
@@ -0,0 +1,319 @@
+    OtelTraceState otelTraceState = OtelTraceState.parse(otelTraceStateString);
+
+    long randomValue;
+    if (otelTraceState.hasValidRandomValue()) {
+      randomValue = otelTraceState.getRandomValue();
+    } else if (isRandomTraceIdFlagSet) {
+      randomValue = OtelTraceState.parseHex(traceId, 18, 14, getInvalidRandomValue());
+    } else {
+      randomValue = randomValueGenerator.generate(traceId);
+      otelTraceState.invalidateThreshold();
+      otelTraceState.setRandomValue(randomValue);
+    }
+
+    long parentThreshold;
+    if (otelTraceState.hasValidThreshold()) {
+      long threshold = otelTraceState.getThreshold();
+      if (((randomValue < threshold) == isParentSampled) || threshold == 0) {
+        parentThreshold = threshold;
+      } else {
+        parentThreshold = getInvalidThreshold();
+      }
+    } else if (isParentSampled) {
+      parentThreshold = getMaxThreshold();
+    } else {
+      parentThreshold = 0;
+    }
+
+    // determine new threshold that is used for the sampling decision
+    long threshold = getThreshold(parentThreshold, isRoot);
+
+    // determine sampling decision
+    boolean isSampled;
+    if (isValidThreshold(threshold)) {
+      isSampled = (randomValue < threshold);
+      if (0 < threshold && threshold < getMaxThreshold()) {
+        otelTraceState.setThreshold(threshold);
+      } else {
+        otelTraceState.invalidateThreshold();
+      }
+    } else {
+      isSampled = isParentSampled;
+      otelTraceState.invalidateThreshold();
+    }
+
+    SamplingDecision samplingDecision =
+        isSampled ? SamplingDecision.RECORD_AND_SAMPLE : SamplingDecision.DROP;
+
+    String newOtTraceState = otelTraceState.serialize();
+
+    return new SamplingResult() {
+
+      @Override
+      public SamplingDecision getDecision() {
+        return samplingDecision;
+      }
+
+      @Override
+      public Attributes getAttributes() {
+        return Attributes.empty();
+      }
+
+      @Override
+      public TraceState getUpdatedTraceState(TraceState parentTraceState) {
+        return parentTraceState.toBuilder()
+            .put(OtelTraceState.TRACE_STATE_KEY, newOtTraceState)
+            .build();
+      }
+    };
+  }
+
+  /**
+   * Returns the threshold that is used for the sampling decision.
+   *
+   * <p>NOTE: In future, further information like span attributes could be also added as arguments
+   * such that the sampling probability could be made dependent on those extra arguments. However,
+   * in any case the returned threshold value must not depend directly or indirectly on the random
+   * value. In particular this means that the parent sampled flag must not be used for the
+   * calculation of the threshold as the sampled flag depends itself on the random value.
+   *
+   * @param parentThreshold is the threshold (if known) that was used for a consistent sampling
+   *     decision by the parent
+   * @param isRoot is true for the root span
+   * @return this Builder
```

Comment:
return the threshold to be used for the sampling decision

Reply:
resolved by 1c41d25a5637b27d2ca5f367b20921fee297bf5d